Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,610 @@
+    /// - Returns: A new problem that suggests that the developer removes the duplicated parameter documentation.
+    private func makeDuplicateParameterProblem(_ parameter: Parameter, previous: Parameter) -> Problem {
+        let notes: [DiagnosticNote]
+        if let previousRange = previous.range, let source = previousRange.source {
+            notes = [DiagnosticNote(source: source, range: adjusted(previousRange), message: "Previously documented here")]
+        } else {
+            notes = []
+        }
+        
+        return Problem(
+            diagnostic: Diagnostic(
+                source: parameter.range?.source,
+                severity: .warning,
+                range: adjusted(parameter.range),
+                identifier: "org.swift.docc.DuplicateParameterDocumentation",
+                summary: "Parameter \(parameter.name.singleQuoted) is already documented",
+                notes: notes
+            ),
+            possibleSolutions: [
+                Solution(
+                    summary: "Remove duplicate parameter documentation",
+                    replacements: parameter.range.map { [Replacement(range: adjusted($0), replacement: "")] } ?? []
+                )
+            ]
+        )
+    }
+    
+    /// Creates a new problem about a parameter that's missing documentation.
+    ///
+    /// ## Example
+    /// 
+    /// ```swift
+    /// /// - Paramaters:
+    /// ///   - firstValue: Description of the first parameter
+    /// ///                                                   ^
+    /// ///                                                   Parameter 'secondValue' is missing documentation
+    /// func doSomething(with firstValue: Int, and secondValue: Int) {}
+    /// ```
+    /// 
+    /// - Parameters:
+    ///   - name: The (non-external) name of the undocumented parameter
+    ///   - nextParameter: The next documented parameter for this symbol or `nil`, if the undocumented parameter is the last parameter for the symbol.
+    ///   - standalone: `true` if the existing documented parameters use the standalone syntax `- Parameter someValue:` or `false` if the existing documented parameters use the `- someValue` syntax within a `- Parameters:` list.
+    ///   - location: The end of the last parameter. Used as the diagnostic location when the undocumented parameter is the last parameter of the symbol.
+    /// - Returns: A new problem that suggests that the developer adds documentation for the parameter.
+    private func makeMissingParameterProblem(name: String, before nextParameter: Parameter?, standalone: Bool, lastParameterEndLocation: SourceLocation?) -> Problem {
+        let solutions: [Solution]
+        if let insertLocation = nextParameter?.range?.lowerBound ?? lastParameterEndLocation {
+            let extraWhitespace = "\n///" + String(repeating: " ", count: (nextParameter?.range?.lowerBound.column ?? 1 + (standalone ? 0 : 2) /* indent items in a parameter outline by 2 spaces */) - 1)
```

Comment:
Why include `///` here? Does that make the replacement ("Fix It" in the Xcode UI?) look better?

Reply:
Because the fix it inserts the placeholder text for the new parameter on its own line we need to include "///" after the newline so that the new line is part of the documentation comment.

For example, consider this declaration and documentation comment where the "first" parameter is undocumented and should be inserted before the "second".

```swift
/// - Paramaters:
///   - second: Some description of the first parameter
func someFunction(first: Int, second: Int) {}
```

Inserting only `- first: placeholder` without a newline would result in 

```swift
/// - Paramaters:
///   - first: placeholder- second: Some description of the first parameter
func someFunction(first: Int, second: Int) {}
```

Inserting only `- parameterName: placeholder\n` with a newline would result in 

```swift
/// - Paramaters:
///   - first: placeholder
- second: Some description of the first parameter
func someFunction(first: Int, second: Int) {}
```

But inserting `- parameterName: placeholder\n///   ` with a newline, a triple slash, and the same leading whitespace as the previous parameter had would result in 

```swift
/// - Paramaters:
///   - first: placeholder
///   - second: Some description of the first parameter
func someFunction(first: Int, second: Int) {}
```